src/test/java/org/jabref/logic/importer/fetcher/GrobidCitationFetcherTest.java

```diff
@@ -100,4 +107,17 @@ public void grobidPerformSearchWithInvalidDataTest(String invalidInput) {
         assertEquals(Collections.emptyList(), entries);
     }
 
+    @Test
+    public void performSearchThrowsExceptionInCaseOfConnectionIssues() throws IOException {
+        GrobidService grobidServiceMock = mock(GrobidService.class);
+        when(grobidServiceMock.processCitation(anyString(), any())).thenThrow(new IOException("Any IO Exception"));
+        grobidCitationFetcher = new GrobidCitationFetcher(importFormatPreferences, grobidServiceMock);
+
+        try {
+            grobidCitationFetcher.performSearch("any text");
+            fail("performSearch should throw an UncheckedIOException, when there are underlying IOException.");
+        } catch (UncheckedIOException expected) {
```

Comment:
Very good that you added a test as well. Small suggestion for improvement: use `Assertions.assertThrows` to ensure that the exception is indeed thrown.

Reply:
Will have a look. Thank you.